src/main/java/org/mockito/internal/configuration/plugins/PluginRegistry.java

```diff
@@ -50,10 +59,36 @@ MockMaker getMockMaker() {
      * Returns the instantiator provider available for the current runtime.
      *
      * <p>Returns {@link org.mockito.internal.creation.instance.DefaultInstantiatorProvider} if no
-     * {@link org.mockito.plugins.InstantiatorProvider} extension exists or is visible in the current classpath.</p>
+     * {@link org.mockito.plugins.InstantiatorProvider2} extension exists or is visible in the
+     * current classpath.</p>
      */
-    InstantiatorProvider getInstantiatorProvider() {
-      return instantiatorProvider;
+    InstantiatorProvider2 getInstantiatorProvider() {
+        // Lazily init defaultInstantiatorProvider as all dependencies of the default instantiator
```

Comment:
To avoid problems with concurrency, we prefer to instantiate all plugins statically vs. lazily. Can you explain why you suggest doing it lazily here?

Reply:
By using the loadPlugin with two parameters, this can be avoided.